dvc/parsing/interpolate.py

```diff
@@ -80,6 +81,31 @@ def _(obj: bool):
     return "true" if obj else "false"
 
 
+@to_str.register(dict)
+def _(obj: dict):
+    result = ""
+    for k, v in flatten(obj).items():
+        if isinstance(v, bool):
+            if v:
+                result += f"--{k} "
+            else:
+                result += f"--no-{k} "
+        elif isinstance(v, str):
+            result += f"--{k} '{v}' "
+        elif isinstance(v, Iterable):
+            for i in v:
+                if isinstance(i, str):
+                    i = f"'{i}'"
+                elif isinstance(i, Iterable):
+                    raise ParseError(
+                        f"Cannot interpolate nested iterable in '{k}'"
+                    )
+                result += f"--{k} {i} "
```

Comment:
```python
> import argparse
> parser = argparse.ArgumentParser()
> parser.add_argument('--foo', nargs='*')
> parser.parse_args('--foo x y'.split())
Namespace(foo=['x', 'y'])

> parser.parse_args('--foo x --foo y'.split())
Namespace(foo=['y'])
```

Looks like setting arguments for multi times will overwrite the previous one.

Reply:
Let's pick one behavior for now. Is it possible to see which one is more widely used in existing ML CLIs? I would guess the `nargs` `--foo 1 2` behavior is more common, even though I find the explicit `append` behavior easier to read.

Reply:
Added `config.parsing.list` with `nargs` and `append` options. It doesn't look that bad to have the options.

Reply:
Just noticed this, it's a bad idea to make `dvc.yaml` parsing conditional, it should be self-contained.